airflow/jobs.py

```diff
@@ -1434,7 +1434,7 @@ def _execute_helper(self, processor_manager):
                                                           State.NONE)
 
                 self._execute_task_instances(simple_dag_bag,
-                                             (State.SCHEDULED,))
```

Comment:
hmm do you have any thoughts on how to avoid this race condition then? the problem seems to be when it does this second check at https://github.com/apache/incubator-airflow/blob/master/airflow/models.py#L1290 that the task is turned to queued but it isn't actually added to the scheduler queue (like it is when it is marked as QUEUED by the primary check at https://github.com/apache/incubator-airflow/blob/master/airflow/jobs.py#L1111). so nothing ever picks up the task (unless you restart the scheduler) and then the dag never finishes. the test case I added below (if you run it manually against local executor) will exhibit this behavior 100% of the time.